packages/nimble-components/src/table-column/table-column-text.ts

```diff
@@ -0,0 +1,81 @@
+import { attr, css, html, ViewTemplate } from '@microsoft/fast-element';
+import { DesignSystem } from '@microsoft/fast-foundation';
+import type { StringField, TableCellState } from '../table/types';
+import { TableColumn } from './table-column';
+import {
+    bodyFontColor,
+    bodyFont,
+    controlLabelFontColor
+} from '../theme-provider/design-tokens';
+
+type TableColumnTextCellData = StringField<'value'>;
+interface TableColumnTextColumnConfig {
+    placeholder: string;
+}
+
+/**
+ * The table column for displaying strings.
+ */
+export class TableColumnText extends TableColumn<
+TableColumnTextCellData,
+TableColumnTextColumnConfig
+> {
+    public cellStateDataFieldNames = ['value'] as const;
+
+    @attr({ attribute: 'field-name' })
+    public fieldName!: string;
+
+    @attr
+    public placeholder?: string;
+
+    public override readonly cellStyles = css`
+        .placeholder {
+            font: ${bodyFont};
+            color: ${controlLabelFontColor};
+            white-space: nowrap;
+        }
+
+        .text-value {
+            font: ${bodyFont};
+            color: ${bodyFontColor};
+            white-space: nowrap;
+        }
+    `;
+
+    public readonly cellTemplate: ViewTemplate<
+    TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+    > = html<
+        TableCellState<TableColumnTextCellData, TableColumnTextColumnConfig>
+        >`
+        <span
+            class="${x => (x.data.value !== null && x.data.value !== undefined
+        ? 'text-value'
+        : 'placeholder')}"
+        >
+            ${x => (x.data.value !== null && x.data.value !== undefined
+        ? x.data.value
+        : x.columnConfig.placeholder)}
+        </span>
+    `;
+
+    public getColumnConfig(): TableColumnTextColumnConfig {
+        return { placeholder: this.placeholder ?? '' };
+    }
+
+    public getRecordFieldNames(): string[] {
+        if (!this.fieldName) {
+            throw new Error('valueKey must be provided.');
+        }
+        return [this.fieldName];
+    }
+
+    public validateCellData(_: TableColumnTextCellData): void {}
+}
+
+const nimbleTableColumnText = TableColumnText.compose({
```

Comment:
Unlike cell and row this is a user facing custom element and my instinct is to treat is like one. Particularly since it has styles, a template, and tests. I think it's reasonable to expect that it has it's own stories and matrix stories as well distinct from the table.

I'd advocate that it is organized like another component would be in:
```
/src/table-column-text/index.ts
/src/table-column-text/template.ts
/src/table-column-text/styles.ts
/src/table-column-text/tests/*.ts
```

Maybe the TableColumn type can just go under `/src/table/models/table-column` which was recently added to our folder structure doc: https://github.com/ni/nimble/blob/main/packages/nimble-components/CONTRIBUTING.md#folder-structure

Reply:
The only reason that I (and @jattasNI I believe) was skeptical about this organization was that there could be a decent amount of nimble-defined columns, and having them all be organized at the top-level seemed a bit like table pollution. That said, I think you are right in that the column custom elements will need to have styles and templates (and they need their own files) then having all of the column components defined in a single directory seems problematic (assuming we use index.ts for the implementation).

Another proposal could be something like this:
```

Reply:
I buy that. To reduce verbosity a tad and bringing in the other discussion about the base folder:

```
/src/table-columns/base/*
/src/table-columns/text/*
```

Or maybe remove the pluralization since it'll make the class names align better (other than base):

```
/src/table-column/base/[index|style|template].ts
/src/table-column/text/[index|style|template].ts
```

Reply:
Used your last suggestion.